File: androidApp/src/main/java/com/luciddreaming/android/ui/screens/SettingsScreen.kt
package com.luciddreaming.android.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.luciddreaming.android.viewmodel.SettingsViewModel

@Composable
fun SettingsScreen(
    viewModel: SettingsViewModel,
    onUnbindSuccess: () -> Unit,
    paddingValues: PaddingValues
) {
    val settings by viewModel.settings.collectAsState()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(paddingValues)
            .padding(16.dp)
    ) {
        Text(
            text = "Settings",
            style = MaterialTheme.typography.titleLarge,
            modifier = Modifier.padding(bottom = 32.dp)
        )

        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp),
            elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp)
            ) {
                Text(
                    text = "Refresh Settings",
                    style = MaterialTheme.typography.titleMedium,
                    modifier = Modifier.padding(bottom = 16.dp)
                )

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(text = "Auto Refresh")
                    Switch(
                        checked = settings.autoRefresh,
                        onCheckedChange = {
                            viewModel.updateAutoRefresh(it)
                        }
                    )
                }

                Spacer(modifier = Modifier.height(16.dp))

                Text(
                    text = "Refresh Interval: ${settings.refreshInterval} seconds",
                    modifier = Modifier.padding(bottom = 8.dp)
                )
                Slider(
                    value = settings.refreshInterval.toFloat(),
                    onValueChange = { newValue ->
                        viewModel.updateRefreshInterval(newValue.toInt())
                    },
                    valueRange = 5f..30f,
                    steps = 5,
                    modifier = Modifier.fillMaxWidth()
                )
            }
        }

        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp),
            elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp)
            ) {
                Text(
                    text = "Appearance",
                    style = MaterialTheme.typography.titleMedium,
                    modifier = Modifier.padding(bottom = 16.dp)
                )

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(text = "Dark Mode")
                    Switch(
                        checked = settings.darkMode,
                        onCheckedChange = {
                            viewModel.updateDarkMode(it)
                        }
                    )
                }
            }
        }

        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp),
            elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp)
            ) {
                Text(
                    text = "Connection",
                    style = MaterialTheme.typography.titleMedium,
                    modifier = Modifier.padding(bottom = 16.dp)
                )

                Button(
                    onClick = {
                        viewModel.disconnect()
                        onUnbindSuccess()
                    },
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Text("Disconnect")
                }
            }
        }

        Button(
            onClick = {
                viewModel.resetSettings()
            },
            modifier = Modifier.fillMaxWidth()
        ) {
            Text("Reset Settings")
        }
    }
}

File: iosApp/LucidDreaming/UI/SettingsScreen.swift
import SwiftUI

struct SettingsScreen: View {
    @ObservedObject var viewModel: SettingsViewModel
    let onDisconnected: () -> Void
    
    var body: some View {
        ScrollView {
            VStack(spacing: 20) {
                Text("Settings")
                    .font(.title)
                    .fontWeight(.bold)
                
                VStack(spacing: 15) {
                    Text("Refresh Settings")
                        .font(.headline)
                    
                    Toggle(isOn: Binding(
                        get: { viewModel.settings.autoRefresh },
                        set: { viewModel.updateAutoRefresh(enabled: $0) }
                    )) {
                        Text("Auto Refresh")
                    }
                    
                    Text("Refresh Interval: \(viewModel.settings.refreshInterval) seconds")
                        .font(.body)
                    
                    Slider(
                        value: Binding(
                            get: { Double(viewModel.settings.refreshInterval) },
                            set: { viewModel.updateRefreshInterval(interval: Int($0)) }
                        ),
                        in: 5...30,
                        step: 5
                    )
                }
                .padding()
                .background(Color.gray.opacity(0.1))
                .cornerRadius(10)
                
                VStack(spacing: 15) {
                    Text("Appearance")
                        .font(.headline)
                    
                    Toggle(isOn: Binding(
                        get: { viewModel.settings.darkMode },
                        set: { viewModel.updateDarkMode(enabled: $0) }
                    )) {
                        Text("Dark Mode")
                    }
                }
                .padding()
                .background(Color.gray.opacity(0.1))
                .cornerRadius(10)
                
                VStack(spacing: 15) {
                    Text("Connection")
                        .font(.headline)
                    
                    Button(action: {
                        viewModel.disconnect()
                        onDisconnected()
                    }) {
                        Text("Disconnect")
                            .padding()
                            .background(Color.red)
                            .foregroundColor(.white)
                            .cornerRadius(8)
                            .frame(maxWidth: .infinity)
                    }
                }
                .padding()
                .background(Color.gray.opacity(0.1))
                .cornerRadius(10)
                
                Button(action: {
                    viewModel.resetSettings()
                }) {
                    Text("Reset Settings")
                        .padding()
                        .background(Color.orange)
                        .foregroundColor(.white)
                        .cornerRadius(8)
                        .frame(maxWidth: .infinity)
                }
            }
            .padding()
        }
    }
}
